x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/processors/grok/grok_ai_suggestions.tsx

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import {
+  EuiBadge,
+  EuiButton,
+  EuiButtonIcon,
+  EuiCallOut,
+  EuiCodeBlock,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+  EuiPopover,
+  EuiText,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useWatch, useFormContext } from 'react-hook-form';
+import { useAbortController } from '@kbn/observability-utils-browser/hooks/use_abort_controller';
+import { RecursiveRecord, StreamDefinition } from '@kbn/streams-schema';
+import type { FindActionResult } from '@kbn/actions-plugin/server';
+import { UseGenAIConnectorsResult } from '@kbn/observability-ai-assistant-plugin/public/hooks/use_genai_connectors';
+import { useKibana } from '../../../../hooks/use_kibana';
+import { GrokFormState, ProcessorFormState } from '../../types';
+import { UseProcessingSimulatorReturn } from '../../hooks/use_processing_simulator';
+
+const RefreshButton = ({
+  generatePatterns,
+  connectors,
+  selectConnector,
+  currentConnector,
+}: {
+  generatePatterns: () => void;
+  selectConnector?: UseGenAIConnectorsResult['selectConnector'];
+  connectors?: FindActionResult[];
+  currentConnector?: string;
+}) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const splitButtonPopoverId = useGeneratedHtmlId({
+    prefix: 'splitButtonPopover',
+  });
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  return (
+    <EuiFlexGroup responsive={false} gutterSize="xs" alignItems="center">
+      <EuiFlexItem grow={false}>
+        <EuiButton
+          size="s"
+          data-test-subj="streamsAppGrokAiSuggestionsRefreshSuggestionsButton"
+          onClick={generatePatterns}
+          disabled={currentConnector === undefined}
+        >
+          {i18n.translate(
+            'xpack.streams.streamDetailView.managementTab.enrichment.processorFlyout.refreshSuggestions',
+            {
+              defaultMessage: '✨ Generate patterns',
```

Comment:
you can also consider using `sparkles` as the icon for the button, that might work better with different themes / dark mode/accessibility etc